Include/internal/pycore_bytesobject.h

```diff
@@ -33,6 +33,44 @@ _PyBytes_ReverseFind(const char *haystack, Py_ssize_t len_haystack,
                      const char *needle, Py_ssize_t len_needle,
                      Py_ssize_t offset);
 
+#include <string.h>
+
+/** Helper function to implement the inplace repeat method on a buffer
+ *
+ * This method repeately doubles the number of bytes copied to reduce to the number of invocations of memcpy
+ */
+static inline void _PyBytes_RepeatInPlace_helper(char* buffer, Py_ssize_t start_len, Py_ssize_t end_len)
+{
+    Py_ssize_t copied = start_len;
+    while (copied < end_len) {
+        Py_ssize_t bytes_to_copy = Py_MIN(copied, end_len - copied);
+        memcpy(buffer + copied, buffer, bytes_to_copy);
+        copied += bytes_to_copy;
+    }
+}
+
+// Helper function to implement the inplace repeat method on a buffer
+static inline void _PyBytes_RepeatInPlace(char* buffer, Py_ssize_t start_len, Py_ssize_t end_len)
+{
+    if (start_len == 1)
+        memset(buffer, buffer[0], end_len);
+    else
+        _PyBytes_RepeatInPlace_helper(buffer, start_len, end_len);
+}
+
+// Helper function to implement the repeat method on a buffer
+static inline void _PyBytes_Repeat(char* dest, Py_ssize_t len_dest, const char* src, Py_ssize_t len_src)
+{
+    if (len_src == 1)
+        memset(dest, src[0], len_dest);
+    else {
+        if (len_dest >= len_src) {
```

Comment:
Should this be an `assert`? Silently doing nothing seems incorrect.

Reply:
Code is correct, but indeed tricky to read. moved the condition to the start of the function.